marshmallow_dataclass/__init__.py

```diff
@@ -161,9 +163,17 @@ def decorator(clazz: Type[_U]) -> Type[_U]:
     return decorator(_cls) if _cls else decorator
 
 
-def class_schema(
-    clazz: type, base_schema: Optional[Type[marshmallow.Schema]] = None
-) -> Type[marshmallow.Schema]:
+@typing.overload
+def class_schema(clazz: type, base_schema: None) -> Type[marshmallow.Schema]:
+    ...
+
+
+@typing.overload
+def class_schema(clazz: type, base_schema: T_SchemaType) -> T_SchemaType:
```

Comment:
Is this annotation correct though ? The object returned by this function is a class that inherits from T_SchemaType, not T_SchemaType itself...

Reply:
This function should return the same type that sanded throw `base_schema`. If `base_schema` is None then return default type.

Reply:
Yes, I understand the problem with the two different overloads for this function. However, I don't think the type annotation you added is correct. The type of base_schema and the output type of the function are not the same.